ocaml/networkd/bin_db/networkd_db.ml

```diff
@@ -31,60 +31,59 @@ let _ =
     (Printf.sprintf "Usage: %s [-bridge <bridge> | -iface <interface>]" name) ;
   try
     let config = Network_config.read_config () in
-    if !bridge <> "" then
-      if List.mem_assoc !bridge config.bridge_config then
-        let bridge_config = List.assoc !bridge config.bridge_config in
-        let ifaces =
-          List.concat_map (fun (_, port) -> port.interfaces) bridge_config.ports
-        in
-        let mac_of_iface ~order name =
-          match List.find_opt (fun dev -> dev.name = name) order with
-          | Some dev ->
-              Ok (Macaddr.to_string dev.mac)
-          | None ->
-              Error (Printf.sprintf "Could not find MAC address of %s\n" name)
-        in
-        let macs =
-          match config.interface_order with
-          | Some order ->
-              List.map (mac_of_iface ~order) ifaces
-              |> List.fold_left
-                   (fun acc mac ->
-                     match (acc, mac) with
-                     | Ok acc, Ok mac ->
-                         Ok (mac :: acc)
-                     | Ok _, Error msg ->
-                         Error msg
-                     | Error msg, _ ->
-                         Error msg
-                   )
-                   (Ok [])
-          | None ->
-              if ifaces <> [] then
-                (* Fallback to use the bridge MAC address when the interface_order
-                   is not available. This can work only because the host installer
-                   requires only one network interface to setup its own networking so far. *)
-                Ok (Option.to_list bridge_config.bridge_mac)
-              else (* No ifaces, no hwaddrs. *)
-                Ok []
-        in
-        match macs with
-        | Error msg ->
-            rc := 1 ;
-            Printf.fprintf stderr "%s\n" msg
-        | Ok macs -> (
-            Printf.printf "interfaces=%s\n" (String.concat "," ifaces) ;
-            Printf.printf "hwaddrs=%s\n" (String.concat "," (List.rev macs)) ;
-            match bridge_config.vlan with
-            | None ->
-                ()
-            | Some (parent, id) ->
-                Printf.printf "vlan=%d\nparent=%s\n" id parent
+    ( if !bridge <> "" then
+        match List.assoc_opt !bridge config.bridge_config with
+        | Some bridge_config -> (
+            let ifaces =
+              List.concat_map
+                (fun (_, port) -> port.interfaces)
+                bridge_config.ports
+            in
+            let macs =
+              let to_mac ~order name =
+                match List.find_opt (fun dev -> dev.name = name) order with
+                | Some dev ->
+                    Either.Left (Macaddr.to_string dev.mac)
+                | None ->
+                    Either.Right name
+              in
+              match (config.interface_order, ifaces) with
+              | Some order, ifaces ->
+                  let oks, errs = List.partition_map (to_mac ~order) ifaces in
```

Comment:
ifaces here can be empty, which means that it will error out

Reply:
Thank you!
My fault. Didn't run it after the fixup. Now it should work:
```
[root@lcy2-dt30 ~]# ./networkd_db -bridge xapi0
interfaces=
hwaddrs=
vlan=999
parent=xapi1
[root@lcy2-dt30 ~]# ./networkd_db -bridge xapi1
interfaces=eth0,eth1
hwaddrs=0c:c4:7a:e7:b6:de
[root@lcy2-dt30 ~]#
```